pkg/kubelet/disk_manager_test.go

```diff
@@ -51,80 +56,54 @@ func testValidPolicy(t *testing.T) {
 	assert.Error(err)
 }
 
-func testSpaceAvailable(t *testing.T) {
-	policy := testPolicy()
-	mockCadvisor := new(cadvisor.Mock)
+func TestSpaceAvailable(t *testing.T) {
+	assert, policy, mockCadvisor := setUp(t)
 	dm, err := newDiskSpaceManager(mockCadvisor, policy)
-	require.NoError(t, err)
-	const mb = 1024 * 1024
+	assert.NoError(err)
+
 	mockCadvisor.On("DockerImagesFsInfo").Return(cadvisorApi.FsInfo{
-		Usage:    400 * mb,
-		Capacity: 1000 * mb,
+		Usage:     400 * mb,
+		Capacity:  1000 * mb,
+		Available: 600 * mb,
 	}, nil)
 	mockCadvisor.On("RootFsInfo").Return(cadvisorApi.FsInfo{
 		Usage:    9 * mb,
 		Capacity: 10 * mb,
 	}, nil)
-	ok, err := dm.IsDockerDiskSpaceAvailable()
-	require.NoError(t, err)
-	require.True(t, ok)
-	ok, err = dm.IsRootDiskSpaceAvailable()
-	require.NoError(t, err)
-	require.True(t, ok)
-}
 
-func testRootFsAvailable(t *testing.T) {
-	policy := testPolicy()
-	policy.RootFreeDiskMB = 10
-	mockCadvisor := new(cadvisor.Mock)
-	dm, err := newDiskSpaceManager(mockCadvisor, policy)
-	require.NoError(t, err)
+	dm.Unfreeze()
 
-	const mb = 1024 * 1024
-	// 500MB available
-	mockCadvisor.On("DockerImagesFsInfo").Return(cadvisorApi.FsInfo{
-		Usage:     9500 * mb,
-		Capacity:  10000 * mb,
-		Available: 500 * mb,
-	}, nil)
-	// 10MB available
-	mockCadvisor.On("RootFsInfo").Return(cadvisorApi.FsInfo{
-		Usage:     990 * mb,
-		Capacity:  1000 * mb,
-		Available: 10 * mb,
-	}, nil)
 	ok, err := dm.IsDockerDiskSpaceAvailable()
-	require.NoError(t, err)
-	require.True(t, ok)
+	assert.NoError(err)
+	assert.True(ok)
+
 	ok, err = dm.IsRootDiskSpaceAvailable()
-	require.NoError(t, err)
-	require.False(t, ok)
+	assert.NoError(err)
+	assert.False(ok)
 }
 
-func testFsInfoError(t *testing.T) {
-	assert := assert.New(t)
-	policy := testPolicy()
+func TestFsInfoError(t *testing.T) {
+	assert, policy, mockCadvisor := setUp(t)
 	policy.RootFreeDiskMB = 10
-	mockCadvisor := new(cadvisor.Mock)
 	dm, err := newDiskSpaceManager(mockCadvisor, policy)
-	require.NoError(t, err)
+	assert.NoError(err)
 
+	dm.Unfreeze()
 	mockCadvisor.On("DockerImagesFsInfo").Return(cadvisorApi.FsInfo{}, fmt.Errorf("can't find fs"))
 	mockCadvisor.On("RootFsInfo").Return(cadvisorApi.FsInfo{}, fmt.Errorf("EBUSY"))
 	ok, err := dm.IsDockerDiskSpaceAvailable()
 	assert.Error(err)
-	require.True(t, ok)
+	assert.True(ok)
 	ok, err = dm.IsRootDiskSpaceAvailable()
 	assert.Error(err)
-	require.True(t, ok)
+	assert.True(ok)
 }
 
-func testCache(t *testing.T) {
-	policy := testPolicy()
-	mockCadvisor := new(cadvisor.Mock)
+func TestCache(t *testing.T) {
+	assert, policy, mockCadvisor := setUp(t)
 	dm, err := newDiskSpaceManager(mockCadvisor, policy)
-	require.NoError(t, err)
-	const mb = 1024 * 1024
+	assert.NoError(err)
+
```

Comment:
Good catch. I think it's a good idea. This is one of the original tests that I enabled through making the function uppercase. To be honest I noted it worked and moved on. Looks like it could use a bit of work. I'll take a look.